### ISSUE TITLE
Folder named "script" is detected as unlisted dependency

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [webpro/knip#417]
I'm seeing a false positive from the following `package.json` `"scripts"` entry:

```plaintext
"test:migrate": "vitest run -r script/",
```

If I rename the folder to another name like `asdf`, Knip doesn't complaint. But with specifically `script/`:

```plaintext
> create-typescript-app@1.52.2 lint:knip /home/runner/work/create-typescript-app/create-typescript-app
> knip

The CJS build of Vite's Node API is deprecated. See https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.
Unlisted dependencies (1)
script  package.json
 ELIFECYCLE  Command failed with exit code 1.
```

Repro available on https://github.com/JoshuaKGoldberg/create-typescript-app/pull/1164 -> https://github.com/JoshuaKGoldberg/create-typescript-app/commit/c67544a5189640f2935fcb963229ec228d05a510#diff-48d5ba4681726b42e98dae10c08bd94f7f9836644c8f183c6475d10dcf67ebf1L11 -> https://github.com/JoshuaKGoldberg/create-typescript-app/actions/runs/7372901535/job/20061636940?pr=1164.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v3.10.0. See [Release 3.10.0](https://github.com/webpro/knip/releases/tag/3.10.0) for release notes._

**Comment by webpro** [comment]
Thanks for reporting this, Josh!

The cause was the `-r` argument being part of the default/fallback "binary resolver" (many Node-based CLI programs use this flag). In `vitest` it means something else so that's resolved now. Doesn't matter much, but I got the same results for any folder that does not exist or does not have an `index.*` script file.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v4.0.0-canary.7. See [Release 4.0.0-canary.7](https://github.com/webpro/knip/releases/tag/4.0.0-canary.7) for release notes._

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v4.0.0. See [Release 4.0.0](https://github.com/webpro/knip/releases/tag/4.0.0) for release notes._
